nixos/modules/services/security/proxychains.nix

```diff
@@ -0,0 +1,184 @@
+          type = types.nullOr types.int;
+          default = null;
+          description = ''
+            Chain length for random chain.
+          '';
+        };
+      };
+
+      proxyDNS = mkOption {
+        type = types.bool;
+        default = true;
+        description = "Proxy DNS requests - no leak for DNS data.";
+      };
+
+      quietMode = mkEnableOption "Quiet mode (no output from the library).";
+
+      remoteDNSSubnet = mkOption {
+        type = types.enum [ 10 127 224 ];
+        default = 224;
+        apply = x: "remote_dns_subnet ${builtins.toString x}";
+        description = ''
+          Set the class A subnet number to use for the internal remote DNS mapping, uses the reserved 224.x.x.x range by default.
+        '';
+      };
+
+      tcpReadTimeOut = mkOption {
+        type = types.int;
+        default = 15000;
+        description = "Connection read time-out in milliseconds.";
+        apply = x: "tcp_read_time_out ${builtins.toString x}";
+      };
+
+      tcpConnectTimeOut = mkOption {
+        type = types.int;
+        default = 8000;
+        description = "Connection time-out in milliseconds.";
+        apply = x: "tcp_connect_time_out ${builtins.toString x}";
+      };
+
+      localnet = mkOption {
+        type = types.str;
+        default = "127.0.0.0/255.0.0.0";
+        description = "By default enable localnet for loopback address ranges.";
+      };
+
+      proxies = mkOption {
+        type = types.listOf (types.submodule proxyOptions);
+        description = ''
+          List of proxies to be used by proxychains.
+        '';
+
+        example = literalExample ''
+          [
+            { type = "socks4";
+              host = "127.0.0.1";
+              port = 1337;
+            }
+          ]
+        '';
+
+        apply = xs:
+          let proxyLine = x: "${x.type} ${x.host} ${x.port}";
+          in ''
+            [ProxyList]
+            ${builtins.concatStringsSep "\n" (builtins.map proxyLine xs)}
+          '';
+      };
+
+      extraConfig = mkOption {
+        type = types.lines;
+        default = "";
+        description = ''
+          Extra configuration. Contents will be added verbatim to proxychains
+          configuration file.
+        '';
+      };
+    };
+
+  };
+
+  ###### implementation
+
+  meta.maintainers = with maintainers; [ sorki ];
+
+  config = mkIf cfg.enable {
+
+    assertions = singleton {
+      assertion = cfg.chain.type != "random" && cfg.chain.length == null;
+      message = ''
+        Option `services.proxychains.chain.length`
+        only makes sense with `services.proxychains.chain.type` = "random".
+      '';
+    };
+
+    services.proxychains.proxies = mkIf config.services.tor.client.enable
+      (mkDefault [{
+        type = "socks4";
+        host = "127.0.0.1";
+        port = 9050;
+      }]);
+
+    environment.etc."proxychains.conf".text = configFile;
+    environment.systemPackages = [ pkgs.proxychains ];
```

Comment:
`proxychain` is under `services.*`, but I don't see any service being defined. If it isn't a service it should be under `programs.*` instead.